oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/JwtAccessTokenOAuth2TokenIssuer.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.authorization.token;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.lang.NonNull;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames;
+import org.springframework.security.oauth2.jose.JoseHeader;
+import org.springframework.security.oauth2.jose.jws.SignatureAlgorithm;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.jwt.JwtClaimsSet;
+import org.springframework.security.oauth2.jwt.JwtEncoder;
+import org.springframework.util.Assert;
+
+/**
+ * Implementation of {@link OAuth2TokenIssuer} to return {@link OAuth2AccessToken} with serialized {@link Jwt} value as
+ * token value.
+ *
+ * @see OAuth2TokenIssuer
+ *
+ * @author Alexey Nesterov
+ * @since 0.0.4
+ */
+public class JwtAccessTokenOAuth2TokenIssuer implements OAuth2TokenIssuer<OAuth2AccessToken> {
+
+	private Clock clock = Clock.systemUTC();
+
+	private final JwtEncoder jwtEncoder;
+
+	private JwtOAuth2TokenCustomizer customizer;
+	private Iterable<JwtOAuth2TokenValidator> validators;
+
+	public JwtAccessTokenOAuth2TokenIssuer(JwtEncoder jwtEncoder) {
+		this.jwtEncoder = jwtEncoder;
+		this.customizer = (header, claims, request) -> {};
+		this.validators = getDefaultValidators();
+	}
+
+	void setClock(Clock clock) {
+		this.clock = clock;
+	}
+
+	/**
+	 * Override default token validators with the provided collection. Validators are run after all customization applied.
+	 *
+	 * @param validators new validators collection
+	 */
+	public void setValidators(Iterable<JwtOAuth2TokenValidator> validators) {
```

Comment:
We should try to reuse `OAuth2TokenValidator` instead of creating a new type.

Reply:
I made this switch but found that there isn't much to validate really - lifetime is already validated by `NumbusJwtEncoder`. I can't think of any other validations so I decided to remove it alltogether, let me me know if you want to bring it back without any defaults and purely as a customization point.

Reply:
After having a chat with Joe, we decided to remove validators for now.